src/lib/mongoBackend/connectionOperations.cpp

```diff
@@ -359,6 +377,7 @@ bool collectionUpdate
   {
     connection->update(col.c_str(), q, doc, upsert);
     releaseMongoConnection(connection);
+    TIME_STAT_MONGO_WRITE_WAIT_STOP();
```

Comment:
Not sure what you mean here.
This is very WYSIWYG, the measurements are *exactly* where the calls to the macros are in the code. The START macros measure the time for START and the STOP macros get the difference and records it. 

If you believe the macro calls should be elsewhere, just indicate where. This is your territory ...